### DISCUSSION TITLE
Refresh Livewire component

### DISCUSSION THREAD

**Discussion by vertoo** [livewire/livewire#3195] [1. Help]
How can I "re-init" Livewire component? This question is related to my problems with searchable multi-select. Tried many libraries like TomSelect, SlimSelect, Choices (from this example https://forum.laravel-livewire.com/t/select-and-select-multi/1120) and always was problem. With Choices looks all is ok, but... Form is in modal, when I select some value, close modal, open again this value is still selected. How can I refresh this field?
Tried with children Livewire component as modal form, same problems. Mount is fired one time, x-init too.
In application I have few multi-select fields and this is only one issue when I wish I had used vue.js.

**Comment by vertoo** [comment]
I have answer for one problem, to "re-mount" Livewire sub-component add key and change this key before each open modal. But x-init is still fired one time. Is any solution for something like destroy component?

**Reply by drbornot** [reply]
when a modal is totally closed you can get this status and emit to component an event that let you clear vars like
```
$("#relatedIDModal").on('hidden.bs.modal', function(){
                livewire.emit('cleanVars');
});
//..... In component

protected $listeners = ['cleanVars'];

public function cleanVars()
{
   // do your stuff here
}
```
